### DISCUSSION TITLE
ITead Sonoff Zigbee 3.0 USB Dongle Plus model “ZBDongle-E” based on Silicon Labs EFR32MG21 +20dBm radio MCU now sold for $19.99

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#1024] [General]
FYI, ITead has now also released a new “ZBDongle-E” adapter as an improved replacement for [ITead’s previous barebone Silabs EFR32MG21 adapter](https://community.openhab.org/t/itead-zigbee-3-0-usb-dongle-stick-adapter-based-on-silicon-labs-efr32mg21/115034) and an alternative to [ITead’s TI CC2652P based “Sonoff Zigbee 3.0 USB Dongle Plus”](https://community.openhab.org/t/sonoff-zigbee-3-0-usb-dongle-plus-by-itead-is-based-on-texas-instruments-cc2652p-can-now-be-ordered-for-14-99/126738) (which now been renamed to “ZBDongle-P”):

https://itead.cc/product/zigbee-3-0-usb-dongle/

Here is by the way link to FCC filing/report for the new “ZBDongle-E” (a.k.a. "ZBD-E" ) adapter which contains copy of the user manual + a lot more detailed + many more internal and external pictures:

[https://fccid.io/2APN5ZBD-E](https://fccid.io/2APN5ZBD-E)

[https://fccid.io/2APN5ZBD-E/User-Manual/User-Manual-6012461.pdf](https://fccid.io/2APN5ZBD-E/User-Manual/User-Manual-6012461.pdf)

[https://fccid.io/2APN5ZBD-E/Internal-Photos/Internal-Photos-6012455](https://fccid.io/2APN5ZBD-E/Internal-Photos/Internal-Photos-6012455)

[https://fccid.io/2APN5ZBD-E/External-Photos/External-Photos-6012451](https://fccid.io/2APN5ZBD-E/External-Photos/External-Photos-6012451)

It uses the same [EFR32MG21A020F768IM32 (EFR32 Mighty Gecko Series 2) SoC](https://www.silabs.com/wireless/zigbee/efr32mg21-series-2-socs/device.efr32mg21a020f768im32?tab=specs) used on the first ITead Zigbee 3.0 USB Dongle, as well as on CoolKit Technologies "SM-011 V1.0" module used in th ITead Sonoff ZBBridge Zigbee Bridge and in Tuya ZS3L and Tuya ZSLC5 SoC modules. It includes a 80 MHz ARM Cortex-M33 core, 768 Flash Storage, 64 RAM, in a QFN32 package, which radio has + 20 dBm maximum power output and receive sensitivity of -104 (250 kbps O-QPSK DSSS) dBm. 

![image](https://user-images.githubusercontent.com/6320001/182463678-aeb80d30-d7cc-4dea-bd00-1d5b0209b097.png)

![image](https://user-images.githubusercontent.com/6320001/182463698-86f1b585-0a76-47be-a869-ee4f56dfe433.png)

![image](https://user-images.githubusercontent.com/6320001/182463720-1b0afdab-07dd-478e-9b59-9afd1950ee50.png)

![image](https://user-images.githubusercontent.com/6320001/182463741-bcf6ace4-3756-4b48-81f1-dcd95c3a9221.png)

**Comment by Hedda** [comment]
@Daniel-zhan-itead Do you know or can find out from the ITead engineer who designed it if this new Silicon Labs EFR32 based "ZBDongle-E" USB adapter is also shipping from the factory with its own unique USB "product description id" written to the EEPROM of the CH9102F USB-to-UART chip, (and not using the same as the USB "product description id" as your CC2652P based USB-adapter)? 

I ask as it should really not use the same USB "product description id" that is already in use by the existing Texas Instruments CC2652 based "ZBDongle-P" USB adapter as that should be unique to it so that they can be uniquely identified as separate types of Zigbee adapters.

The main beneficial reason for this is that they are using different Zigbee chips that do not use the same serial interface, as one should be configured to use "ezsp" as radio type for Silicon Labs based radios and the other should use "znp" radio type for Texas Instruments based radios in Home Assistant's ZHA integration, which mean that they should really have different unique USB "product description id" so can be automatically discovered as a different type of adapters that should use different radio types in Home Assistant. 

Please see this discussion on Home Assistant community forum for more information: 

https://community.home-assistant.io/t/unique-friendly-name-description-for-automatic-zigbee-usb-adapter-discovery-in-home-assistant-zha-using-dongle-vendor-product-ids/337077

A follow-up question to that is if you could arrange for that ITead engineer to submit a pull request to Home Assistant core repository with the new unique USB "product description id" written to the EEPROM of the CH9102F chip on the new "ZBDongle-E" adapter to add automatic discovery, please see the similar request here: 

https://github.com/home-assistant/core/pull/62171

PS: @Daniel-zhan-itead Also wondering if you know if ITead has shipped hardware samples of this new dongle to lead zigpy developers?

PPS: @Daniel-zhan-itead maybe consider shipping  hardware samples of this new dongle to ZHA, Domoticz, and Jeedom developers too? ;)

**Reply by Daniel-zhan-itead** [reply]
@Hedda
The  "ZBDongle-E" USB "product description id" has been written of SONOFF Zigbee 3.0 USB Dongle Plus V2 to EEPROM from the device, but we also found that the dongle can not be auto-discovered in HA.
As you advised, we will submit a pull request next week to update the "product description id" so that users can add dongles with automatic discovery.

Refer to samples for developers, I will try to contact them to see if they are willing to test and review the dongle.

**Comment by pipiche38** [comment]
side question, any chance to purchase it in Europe ? purpose is to confirm the compatibility with Domoticz

**Reply by Daniel-zhan-itead** [reply]
@pipiche38 I can send you the sample to test if you want, please contact me by mail.
daniel.zhan@itead.cc

**Reply by pipiche38** [reply]
thanks @Daniel-zhan-itead , mail sent

**Reply by Hedda** [reply]
> any chance to purchase it in Europe ?

@pipiche38 FYI , I live in Europe myself and I have ordered directly from ITead’s official China website at https://itead.cc several times without any problems. They recently even made that simpler ordering directly fron China by allowing you pre-pay your custom/import-fees and VAT/tax in advance if you choose DHL as shipping option. Shipping cost with DHL is however more expensive so tip there is to co-order with a few friends if only orderer Zigbee USB dongles or make a larger order yourself if also buying Zigbee devices. I can personally recommend all their battery operated devices.

**Reply by pipiche38** [reply]
@Daniel-zhan-itead , I'm using the zigpy/bellows library and here are some feedbacks and especially it would be good if a Radio manufacturer and radio board names could be given ;-) As they look empty for now, while the EmberZNet version is 6.10.3.0 build 297

```

2022-08-22 20:29:11,445 DEBUG   :Sending: b'211b21a95f2a148cfc7e'
2022-08-22 20:29:11,463 DEBUG   :Data frame: b'131ba1a95f2a054da66bb5da55aa6db663b1d854123198c9017e'
2022-08-22 20:29:11,472 DEBUG   :Sending: b'82503a7e'
2022-08-22 20:29:11,476 DEBUG   :Application frame 11 (getMfgToken) received: b'10ffffffffffffffffffffffffffffffff'
2022-08-22 20:29:11,479 DEBUG   :Read MFG_STRING token: b'\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff'
2022-08-22 20:29:11,486 DEBUG   :Send command getMfgToken: (<EzspMfgTokenId.MFG_BOARD_NAME: 2>,)
2022-08-22 20:29:11,491 DEBUG   :Sending: b'327d3821a95f2a179d867e'
2022-08-22 20:29:11,503 DEBUG   :Data frame: b'2418a1a95f2a054da66bb5da55aa6db663b1d85412319804e37e'
2022-08-22 20:29:11,504 DEBUG   :Sending: b'83401b7e'
2022-08-22 20:29:11,507 DEBUG   :Application frame 11 (getMfgToken) received: b'10ffffffffffffffffffffffffffffffff'
2022-08-22 20:29:11,511 DEBUG   :Read MFG_BOARD_NAME token: b'\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff'
2022-08-22 20:29:11,518 DEBUG   :Send command getValue: (<EzspValueId.VALUE_VERSION_INFO: 17>,)
2022-08-22 20:29:11,522 DEBUG   :Sending: b'431921a9fe2a044cd87e'
2022-08-22 20:29:11,534 DEBUG   :Data frame: b'3519a1a9fe2a15b570954c2fa95538f1fd7e'
2022-08-22 20:29:11,536 DEBUG   :Sending: b'8430fc7e'
2022-08-22 20:29:11,539 DEBUG   :Application frame 170 (getValue) received: b'00072901060a0300aa'
2022-08-22 20:29:11,549 DEBUG   :EZSP Radio manufacturer:
2022-08-22 20:29:11,550 DEBUG   :EZSP Radio board name:
2022-08-22 20:29:11,552 DEBUG   :EmberZNet version: 6.10.3.0 build 297
```

**Reply by MattWestb** [reply]
Writing manufacture tokens  in EFR32 user data: https://github.com/MattWestb/EFR32-FW/tree/main/BrandingEFR32 or reading Silabs papers.

**Reply by Daniel-zhan-itead** [reply]
@pipiche38 Thanks for your suggestion, we will update it. :)

**Reply by Hedda** [reply]
@Daniel-zhan-itead For your information, there is now also another brand new related feature called "hardware integrations" in Home Assistant which will make the next release that I understand able to display additional information about the USB dongle hardware the user has attached under the hardware integration which is shown in Home Assistant's Settings (under "Settings" -> "System" -> "Hardware"). Suggest that ITead/Sonoff submit patches to Home Assistant core with such hardware integrations to support your "ZBDongle-E" and "ZBDongle-P" USB adapters in this feature would make your USB dongles become an even more user-friendly experience in Home Assistant. See:

https://github.com/home-assistant/core/pull/76795

This depends "models.py" under 'homeassistant/components/hardware/' but that you should not have to modify:

https://github.com/home-assistant/core/blob/dev/homeassistant/components/hardware/models.py

instead, you add additional hardware integrations similar to proof of concept hardware integration "homeassistant_sky_connect" `(DONGLE_NAME = "Home Assistant Sky Connect"):`

https://github.com/home-assistant/core/tree/dev/homeassistant/components/homeassistant_sky_connect

That is, this is is a separate unrelated new feature from the USB discovery feature for ZHA, so you would have to submit separate pull requests to Home Assistant core for new "hardware integration components for "ITead_Sonoff_ZBDongle-E_USB_hardware" and "ITead_Sonoff_ZBDongle-P_USB_hardware" which should look similar to that "homeassistant_sky_connect" proof of concept hardware integration for a supported USB adapter.

https://github.com/home-assistant/core/tree/dev/homeassistant/components/homeassistant_sky_connect

Before they only had the computer hardware under Settings -> System -> Hardware (like Raspberry Pi, Home Assistant Blue / Hardkernel Odroid-N2, etc.) but now they added the ability to show supported USB adapters as well. Ex:

![image](https://user-images.githubusercontent.com/6320001/186125437-cb12c929-498e-43e3-aeba-07be4e8bb324.png)

**Comment by Hedda** [comment]
@Daniel-zhan-itead  FYI, it seems like @guozi7788 has been trying during the last week to submit a pull request to Home Assistant core repository which should add USB auto-discover for this new [ITead SONOFF Zigbee 3.0 USB Dongle Plus V2 Model "ZBDongle-E" (EFR32MG21 variant)](https://itead.cc/product/zigbee-3-0-usb-dongle/) Zigbee Coordinator USB adapter but not having much luck with Git/GitHub, see:

https://github.com/home-assistant/core/pull/77016

and

https://github.com/home-assistant/core/pull/77192

Looks like he is trying to force a merge himself to Home Assistant core repository (which he does not have write access to so he can not do that) instead of just waiting for Home Assistant core developers to merge the pull request like he should be doing. Thus it may look like that pull request has been merged to the upstream Home Assistant core repository even though it really has not.

If you are in contact with him then let him know that he will need to submit a new pull request for it but this time tell him not to try to force commit it and instead just wait and leave it for others to merge it to that repository after reviewed and approved.

**Reply by Daniel-zhan-itead** [reply]
@Hedda I have talked with @guozi7788 and he will submit one more PR for the ZBDongle-E USB discovery.
he will also raise a PR to [home-assistant.io](https://github.com/home-assistant/home-assistant.io) for adding the ZBDongle-E to ZHA integration documentation list with the change type of "Added documentation for a new feature I'm adding to Home Assistant (next branch)."

**Reply by Daniel-zhan-itead** [reply]
@Hedda Just update info that the PR of ZBDongle-E USB discovery has been merged to home-assistant/core
https://github.com/home-assistant/core/pull/77523#event-7287885980

**Reply by Hedda** [reply]
> @Hedda Just update info that the PR of ZBDongle-E USB discovery has been merged to home-assistant/core [home-assistant/core#77523 (comment)](https://github.com/home-assistant/core/pull/77523#event-7287885980)

Nice! Yes noticed both https://github.com/home-assistant/core/pull/77523 and HA's ZHA docs PR were merged -> https://github.com/home-assistant/home-assistant.io/pull/23929

Should hopefully mean ZHA in the next/upcoming Home Assistant release (2022.9) will work to auto-discover "ZBDongle-E" over USB.

https://rc.home-assistant.io/blog/2022/08/31/release-20229/

**Comment by Hedda** [comment]
@Daniel-zhan-itead or @MattWestb Do anyone of you happen to have full schematics or detailed pinouts of the “ZBDongle-E” board?

[blakadder](https://github.com/blakadder) was asking which pin is the boot button on ZBDongle-E MG21 connected to?

https://itead.cc/product/zigbee-3-0-usb-dongle/

![image](https://user-images.githubusercontent.com/6320001/194843343-51461821-2ed9-4525-a701-89c5b6b0ab94.png)

I assume he would like to repurpose ZBDongle-E as just a Zigbee module in an ESP32-based DIY gateway using Zigbee2Tasmota (Z2T)?

https://tasmota.github.io/docs/Zigbee/

https://templates.blakadder.com/tube_TUBE013_EFR32.html

https://zigbee.blakadder.com/Sonoff_ZBBridge-P.html

https://templates.blakadder.com/ewelink_ZB-GW03

That is, similar to the style of Zigbee to Ethernet "DIY" remote Zigbee Coordinator

https://github.com/zigpy/zigpy/discussions/584

e.i. a DIY variant of Tube's Gateways and Zigstar LAN Gateway that is connecting ESP32 direct to “ZBDongle-E” board pins via UART/serial:

https://github.com/tube0013/tube_gateways

https://github.com/mercenaruss/zigstar_gateways
